mongock-core/mongock-runner/mongock-runner-core/src/main/java/io/mongock/runner/core/executor/ChangeLogRuntimeImpl.java

```diff
@@ -120,6 +124,29 @@ private String getParameterName(Parameter parameter) {
 
 
   private Constructor<?> getConstructor(Class<?> type) {
-    return type.getConstructors()[0];
+    return findChangeUnitConstructor(type)
+        .orElseGet(() -> findDefaultConstructor(type));
+  }
+
+  private Constructor<?> findDefaultConstructor(Class<?> type) {
+    Supplier<Stream<Constructor<?>>> constructorSupplier = () -> Arrays.stream(type.getConstructors());
+    if (constructorSupplier.get().count() > 1) {
+      logger.warn("Mongock found multiple constructors for changeUnit[{}]. " +
+          "It's recommended to annotate the one you want Mongock to use with @ChangeUnitConstructor. " +
+          "FROM VERSION 6 THIS WILL CAUSE AN ERROR ", type.getName());
+    }
+    return constructorSupplier.get()
+        .findFirst()
+        .orElseThrow(() -> new MongockException("Mongock cannot find a valid constructor for changeUnit[%s]", type.getName()));
+  }
```

Comment:
The use of stream was done to keep a functional style of code for easier readability, It also supports Optionals out of box with functions like findFirst(), orElseThrow()

Since the number of constructors usually is a single digit number, I thought the tradeoff of readability trumped efficiency.

This code also followed the [Object Calisthenics](https://williamdurand.fr/2013/06/03/object-calisthenics/) rules like [Only One Level Of Indentation Per Method](https://williamdurand.fr/2013/06/03/object-calisthenics/#1-only-one-level-of-indentation-per-method) and 
[Don’t Use The ELSE Keyword](https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword)

Please let me know if the change is still recommended.

Reply:
I see your point, but still see more practical this approach. Like it simple 😃

Reply:
okay, cool, will make the required changes

Reply:
done